registry/proxy/proxyblobstore.go

```diff
@@ -67,6 +69,29 @@ func (pbs *proxyBlobStore) copyContent(ctx context.Context, dgst digest.Digest,
 	return desc, nil
 }
 
+func (pbs *proxyBlobStore) doResponse(reqMethod string, ctx context.Context, dgst digest.Digest, writer io.Writer) error {
+	switch reqMethod {
+	case http.MethodHead:
+		desc, err := pbs.remoteStore.Stat(ctx, dgst)
+		if err != nil {
+			return err
+		}
+
+		if w, ok := writer.(http.ResponseWriter); ok {
+			setResponseHeaders(w, desc.Size, desc.MediaType, dgst)
+		}
+		return nil
```

Comment:
Might make more sense to have `fallthrough` here when w is not a `http.ResponseWriter` rather than just do nothing. However a better option may just be to make the argument take in `w` as `http.ResponseWriter` if it is not intended to be used anywhere else.